src/Smart.Design.Razor/TagHelpers/Header/HeaderTagHelper.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Html;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Mvc.TagHelpers;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+using Smart.Design.Razor.TagHelpers.Constants;
+
+namespace Smart.Design.Razor.TagHelpers.Header;
+
+[HtmlTargetElement(TagNames.HeaderTagName)]
+public class HeaderTagHelper : TagHelper
+{
+    private readonly IHeaderHtmlGenerator _headerHtmlGenerator;
+
+    /// <summary>
+    /// Home page url from appsettings context
+    /// </summary>
+    public string HomePageUrl { get; set; }
+
+    /// <summary>
+    /// list of available languages
+    /// </summary>
+    public Dictionary<string, string> LanguagesAndLinks { get; set; } = new Dictionary<string, string>();
+
+    /// <summary>
+    /// Current interface language
+    /// </summary>
+    public string CurrentLanguage { get; set; }
+
+    /// <summary>
+    /// Current user full name
+    /// </summary>
+    public string UserName { get; set; }
+
+    /// <summary>
+    /// Path to the current user avatar
+    /// </summary>
+    public string AvatarPath { get; set; }
+
+    /// <summary>
+    /// Dictionnary of label and links items for the drop down menu
+    /// </summary>
+    public Dictionary<string, string> LabelsAndLinks { get; set; } = new Dictionary<string, string>();
+
+    // <summary>
+    /// Creates a new <see cref="SideMenuTagHelper"/>.
+    /// </summary>
+    /// <param name="sideMenuHtmlGenerator">The service that generates Smart Design side menu with labels and navigation links.</param>
+    public HeaderTagHelper(IHeaderHtmlGenerator headerHtmlGenerator)
+    {
+        _headerHtmlGenerator = headerHtmlGenerator;
+    }
+
+    /// <inheritdoc />
+    public override void Process(TagHelperContext context, TagHelperOutput output)
+    {
+        if (!LanguagesAndLinks.Any())
+        {
+            throw new ArgumentException($"{nameof(LanguagesAndLinks)} cannot be null or empty");
+        }
+
+        if (string.IsNullOrEmpty(UserName))
+        {
+            throw new ArgumentException($"{nameof(UserName)} cannot be null or empty");
+        }
+
+        if (!LabelsAndLinks.Any())
+        {
+            throw new ArgumentException($"{nameof(LabelsAndLinks)} cannot be null or empty");
+        }
+
+        if (string.IsNullOrEmpty(AvatarPath))
+        {
+            AvatarPath = "~/unity/images/default_image.jpg";
+        }
```

Comment:
I am wondering if we should rather put the image in the design library instead (and access it from the "~/Content/images/default_image.jpg"). If we want a custom default image for a specific project we could still do that logic in the path passed to the parameter.
Also, if we add the default value directly in the parameter declaration above, we can remove that if :  
 `public string AvatarPath { get; set; } = "~/unity/images/default_image.jpg"`